modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/expression/MBDecision.java

```diff
@@ -52,8 +78,41 @@ public Expression decisionNot(){
      *          ["!=", null, null]: boolean
      * @return
      */
-    public Expression decisionNotEqual(){
-        return null;
+    private Expression decisionNotEqual() {
+        // validate the arg list
+        if (json.size() != 3) {
+            throwUnexpectedArgumentCount(NOT_EQUALS, 2);
+        }
+        // first arg needs to be the operator
+        final String notEqualsOp = parse.get(json, 0);
+        if (!NOT_EQUALS.equals(notEqualsOp)) {
+            // not a NOT_EQUALS(!=) op
+            throwUnexpectedOperationException(NOT_EQUALS, notEqualsOp);
+        }
+        // get the comparables, treat nulls as literals
+        Object comparable1 = parse.value(json, 1, true);
+        Object comparable2 = parse.value(json, 2, true);
+        // handle nulls first
+        if (comparable1 == null) {
+            // not equal if comp2 is not null
+            return ff.literal(comparable2 != null);
```

Comment:
You won't necessarily be able to know this at parse time, if comparable1 is an expression that may return null, so you'll need to use a function here, not a literal